src/Core/Services/Implementations/OrganizationSponsorshipService.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Threading.Tasks;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.Models.Table;
+using Bit.Core.Repositories;
+using Microsoft.AspNetCore.DataProtection;
+
+namespace Bit.Core.Services
+{
+    public class OrganizationSponsorshipService : IOrganizationSponsorshipService
+    {
+        private const string FamiliesForEnterpriseTokenName = "FamiliesForEnterpriseToken";
+        private const string TokenClearTextPrefix = "BWOrganizationSponsorship_";
+
+        private readonly IOrganizationSponsorshipRepository _organizationSponsorshipRepository;
+        private readonly IOrganizationRepository _organizationRepository;
+        private readonly IUserRepository _userRepository;
+        private readonly IPaymentService _paymentService;
+        private readonly IMailService _mailService;
+
+        private readonly IDataProtector _dataProtector;
+
+        public OrganizationSponsorshipService(IOrganizationSponsorshipRepository organizationSponsorshipRepository,
+            IOrganizationRepository organizationRepository,
+            IUserRepository userRepository,
+            IPaymentService paymentService,
+            IMailService mailService,
+            IDataProtectionProvider dataProtectionProvider)
+        {
+            _organizationSponsorshipRepository = organizationSponsorshipRepository;
+            _organizationRepository = organizationRepository;
+            _userRepository = userRepository;
+            _paymentService = paymentService;
+            _mailService = mailService;
+            _dataProtector = dataProtectionProvider.CreateProtector("OrganizationSponsorshipServiceDataProtector");
+        }
+
+        public async Task<bool> ValidateRedemptionTokenAsync(string encryptedToken)
```

Comment:
I just took a look at that method and this one is different enough where this one doesn't fit in there. That one is setup to validate an expiration time but this one does not have that. It might be a good idea to refactor token validation down the road though.